### DISCUSSION TITLE
New-HTMLTable -AutoSize?

### DISCUSSION THREAD

**Discussion by shivtorov** [EvotecIT/PSWriteHTML#376] [General]
Does anyone know how to make HTMLTable to autosize its columns width based on its content?

I had it working once few days ago, while experimenting with some other features. But I have not saved that code. And now I can't find the correct combination of cmdlets/params to make it work again.

What's missing in the following code for AutoSize to work?

```powershell

$table1 = @'
ID,Name
10,Alice
20,Bob
'@ | ConvertFrom-Csv

New-HTML -Temporary -Online -ShowHTML {
    New-HTMLTable -DataTable $table1 -AutoSize
}
```

**Comment by PrzemyslawKlys** [comment]
Your code is fine, but it will try to fit to the page you have it on. If you push more data it will resize properly.

**Reply by shivtorov** [reply]
The table produced by my code has width 100% of the screen and two very wide columns: 
![image](https://user-images.githubusercontent.com/31003608/231145570-8d55112b-9a62-4f3b-adb8-1840b9aff4c2.png)

I want the table to be compact, like this:
![image](https://user-images.githubusercontent.com/31003608/231150025-ec746a37-1f57-4429-a713-03eaaeb6e14d.png)

**Reply by PrzemyslawKlys** [reply]
```powershell
$table1 = @'
ID,Name
10,Alice
20,Bob
'@ | ConvertFrom-Csv

New-HTML -Online -ShowHTML {
    New-HTMLSection -Invisible {
        New-HTMLContainer -Width 200px {
            New-HTMLTable -DataTable $table1 -AutoSize
        }
    }
}
```

**Reply by shivtorov** [reply]
Thank you @PrzemyslawKlys
Now, I have the combination of cmdlets/params that produces the desired outcome.

In case someone has the same problem, the crucial part of the code to get the desired result:

```powershell
New-HTML -Online -ShowHTML {
    New-HTMLSection {
        New-HTMLContainer -Width auto {
            New-HTMLTable -DataTable $table1 -AutoSize -DisableResponsiveTable
        }
    }
}
```

Full code:

```powershell
$table1 = @'
ID,Name
10,Alice
20,Bob
'@ | ConvertFrom-Csv

New-HTML -Online -ShowHTML {
    New-HTMLSection {
        New-HTMLContainer -Width auto {
            New-HTMLTable -DataTable $table1 -AutoSize -DisableResponsiveTable -DisableInfo -DisablePaging -DisableSearch -HideFooter -DisableSelect -HideButtons
        }
    }
}
```
Result:

![image](https://user-images.githubusercontent.com/31003608/231199641-8b66a928-1238-4502-8b64-ce0edab5024d.png)
